doc/reference/trd-appid.md

```diff
@@ -0,0 +1,396 @@
+  multiple applications. The application is identified by a unique
+  identifier I. This application's application credentials consist of
+  a TBF header containing an ECDSA public key, the unique identifier
+  I, and a signed SHA512 hash of the application binary and unique
+  identifier I, signed with the private key corresponding to the
+  public key in the header. The kernel decides whether to accept a
+  particular public key for verification. The application has a global
+  application identifier is the concatenation of its public key and I.
+
+  1. The application has multiple processes. The application is
+  identified by a public key that is also its global application
+  identifier. The application credentials of each Tock binary of the
+  process consist a TBF header containing the public key and a
+  signature of the SHA512 of the application binary made with the
+  corresponding private key.
+
+An application identifier provides an identity for an application
+binary. It allows the Tock kernel to know about the provenance and
+origin of the binary and make access control or security decisions
+based on this information. For example, a kernel may allow only applications whose
+credentials use a particular trusted public key to access restricted
+functionality, but restrict other applications to use a subset of
+available system calls.
+
+Application identifiers are distinct from process
+identifiers; an application identifier is per-application (persists
+across restarts of a Tock binary, for example), while a process
+identifier identifies a particular execution of that binary. At any
+time on a Tock device, each process has a unique process identifier,
+but they can be re-used over time (like POSIX process identifiers).
+
+As the above examples illustrate, application credentials can vary in
+size and content. The credentials that the kernel will accept depends
+on its use case: certain devices will only accept credentials which
+include a particular public key, while others will accept
+many. Furthermore, the internal format of these credentials can vary.
+Finally, the cryptography used in credentials can vary, either due to
+security policies or certification requirements.
+
+4. Credentials in Tock Binary Format Headers
+===============================
+
+To support credentials in Tock binaries, the Tock Binary Format has
+a `TbfHeaderV2Credentials` header. This header is variable length
+and has two fields:
+
+```rust
+pub struct TbfHeaderV2Credentials {
+    format: TbfHeaderV2CredentialsType,
+    data: &[u8],
+}  
+```
+
+The `TbfHeaderV2CredentialsType` defines the format and size of `data`
+field. A `TbfHeaderV2CredentialsType` value MUST have a fixed
+data size and format. Currently supported values are:
+
+```rust
+pub enum TbfHeaderV2CredentialsType {
+    CleartextID = 0,
+    Rsa3072Key = 1,
+    Rsa4096Key = 2,
+    Rsa3072KeyWithID = 3,
+    Rsa4096KeyWithID = 4,
+}
+```
```

Comment:
These should be in the TBF doc instead of here

Reply:
The TBF doc isn't a TRD. It can change at any time. It's problematic to have a finalized/fixed document depend on something which is variable.

Reply:
The problem is that people will go to the TBF doc for details about the TBF. If this isn't there it will lead to confusion

Reply:
Then it should be replicated here and in the TBF doc? It can't be that we have a finalized document that depends on uncontrolled web page.

Reply:
What do you mean uncontrolled web page? The TBF doc describes the interface between the kernel and application loading. It is one of the most stable parts of the kernel (otherwise app loading will break). It is harder to change then this text.

I'm not saying it can't change, but it's unlikely to see breaking changes as that would greatly affect users